client/src/main/java/com/linecorp/decaton/client/DecatonClientImpl.java

```diff
@@ -91,4 +104,29 @@ public DecatonClientImpl(String topic,
     public void close() throws Exception {
         producer.close();
     }
+
+    private TaskMetadataProto convertToTaskMetadataProto(TaskMetaData overrideTaskMetadata) {
+        final TaskMetadataProto.Builder
+                taskMetadataProtoBuilder = TaskMetadataProto.newBuilder()
+                                                            .setSourceApplicationId(applicationId)
+                                                            .setSourceInstanceId(instanceId);
+        if (timestampSupplier != null) {
```

Comment:
`timestampSupplier` is configurable only through package-private constructor and in public constructor, it is supplied by default, so you don't have to expect it to be null.